### PR TITLE
Added bigquery

### PR DESCRIPTION
**Pull request by ajinkya1986**

### CODE REVIEW

File: cartography/intel/gcp/__init__.py [hunk 1]
```diff
@@ -22,14 +22,14 @@
 logger = logging.getLogger(__name__)
 Resources = namedtuple(
     'Resources', 'compute gke cloudfunction crm_v1 crm_v2 dns storage serviceusage \
-        iam admin cloudfunction apigateway cloudkms cloudrun sql bigtable firestore',
+        iam admin apigateway cloudkms cloudrun sql bigtable firestore',
```
Comment (mpurusottamc) [inline cartography/intel/gcp/__init__.py:25]: do we need to add bigquery in the resources list as well?

File: cartography/intel/gcp/bigquery.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+            raise
+
+
+@timeit
+def transform_bigquey_dataset(bigquery: Resource, datasets: List[Dict], project_id: str) -> List[Resource]:
+    list_dataset = []
+    for dataset in datasets:
+        dataset['id'] = dataset.get('datasetReference', {}).get('datasetId', '')
+        dataset['uniqueId'] = f"projects/{project_id}/datasets/{dataset['id']}"
+        dataset['details'] = get_dataset_info(bigquery, dataset['id'])
```
Comment (mpurusottamc) [inline cartography/intel/gcp/bigquery.py:83]: we should populate consolelink property here

File: cartography/intel/gcp/bigquery.py [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+    # BIGQUERY TABLES
+    for dataset in bigquery_datasets:
+        tables = get_bigquery_tables(bigquery, dataset, project_id, common_job_parameters)
+        bigquery_tables = transform_bigquery_tables(bigquery, dataset, tables, project_id)
+        load_bigquery_tables(neo4j_session, bigquery_tables, gcp_update_tag)
+        label.sync_labels(neo4j_session, bigquery_tables, gcp_update_tag, common_job_parameters, 'bigquerytables', 'GCPBigqueryTables')
+    cleanup_gcp_bigquery(neo4j_session, common_job_parameters)
+
+    toc = time.perf_counter()
+    logger.info(f"Time to process Cloud KMS: {toc - tic:0.4f} seconds")
```
Comment (mpurusottamc) [inline cartography/intel/gcp/bigquery.py:332]: change this line. it says KMS.

**Review by mpurusottamc — CHANGES_REQUESTED** [review]
minor changes needed.

File: cartography/intel/gcp/bigquery.py [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+    """
+    try:
+        tables = []
+        request = bigquery.tables().list(projectId=project_id, datasetId=dataset['id'], maxResults=5000)
+        while request is not None:
+            response = request.execute()
+            if 'tables' in response:
+                tables.extend(response.get('tables', []))
+            request = bigquery.tables().list_next(previous_request=request, previous_response=response)
+        if common_job_parameters.get('pagination', {}).get('bigquery', None):
```
Comment (ganeshjagdale9545) [inline cartography/intel/gcp/bigquery.py:130]: pagination use at dataset level not at table level

File: cartography/intel/gcp/bigquery.py [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+
+    :rtype: NoneType
+    :return: Nothing
+    """
+    tic = time.perf_counter()
+
+    logger.info("Syncing Bigquery for project '%s', at %s.", project_id, tic)
+
+    # BIGQUERY DATASETS
+    datasets = get_bigquery_dataset(bigquery, project_id, common_job_parameters)
```
Comment (ganeshjagdale9545) [inline cartography/intel/gcp/bigquery.py:319]: use pagination here

File: cartography/intel/gcp/__init__.py (hunk 1 above)
Comment (ganeshjagdale9545) [inline cartography/intel/gcp/__init__.py:25]: @sagarnavale @ajinkya1986 can you check for this file code?
i saw most of the code is missing here, for gcp please update the `cloudanix-all` branch with `cloudanix-gcp` branch

File: cartography/intel/gcp/bigquery.py (hunk 2 above)
Comment (ajinkya1986) [reply cartography/intel/gcp/bigquery.py:83]: The console link for dataset and tables are of the following format
`https://console.cloud.google.com/bigquery?referrer=search&project=cloudanix-qa-303120&ws=!1m4!1m3!3m2!1scloudanix-qa-303120!2sdataset1`. How should we implement the console link as ws field is different

Comment (mpurusottamc) [reply cartography/intel/gcp/bigquery.py:83]: if we ignore the `ws` attribute does it fail?

File: cartography/intel/gcp/__init__.py [hunk 6]
```diff
@@ -38,14 +44,15 @@
     dns='dns.googleapis.com',
     crm_v1='cloudresourcemanager.googleapis.com',
     crm_v2='cloudresourcemanager.googleapis.com',
+    cloudfunction='cloudfunctions.googleapis.com',
     cloudkms='cloudkms.googleapis.com',
     cloudrun='run.googleapis.com',
     iam='iam.googleapis.com',
-    admin='admin.googleapis.com',
```
Comment (mpurusottamc) [inline cartography/intel/gcp/__init__.py:44]: i think `admin` is needed.

File: cartography/intel/gcp/bigquery.py [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+            raise
+       
+@timeit
+def transform_bigquery_tables(bigquery: Resource, dataset: Dict, tables: List, project_id: str) -> List[Resource]:
+    list_tables = []
+    for  table in tables:
+        table['id'] = table.get('tableReference', {}).get('tableId', '')
+        table['datasetId'] = dataset['id']
+        table['uniqueId'] = f"projects/{project_id}/datasets/{dataset['id']}/tables/{table['id']}"
+        table['details'] = get_table_info(bigquery, project_id, dataset['id'], table['id'])
```
[the remark this answers is not in the document]
Comment (ajinkya1986) [reply cartography/intel/gcp/bigquery.py:151]: Added comment for future reference

File: cartography/intel/gcp/bigquery.py (hunk 2 above)
Comment (ajinkya1986) [reply cartography/intel/gcp/bigquery.py:83]: Added comment for future

File: cartography/intel/gcp/__init__.py (hunk 6 above)
Comment (ajinkya1986) [reply cartography/intel/gcp/__init__.py]: Added admin

**Review by mpurusottamc — APPROVED** [review]
looks good.

**Pull request merged**